common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -69,6 +69,7 @@ public static class Collation {
      * byte for byte equal. All accent or case-insensitive collations are considered non-binary.
      */
     public final boolean isBinaryCollation;
+    public final boolean isLowercaseCollation;
```

Comment:
I think that the name is misleading. Also it doesn't make much sense to keep a flag that will be 0 for 99% of collations and 1 for only 1. You use this only in `StringType` so can you just say there:

```
def isLcaseCollation: Boolean = collationId == LOWERCASE_COLLATION_ID
```
That being said, maybe even we don't need this and you can just make this check in `StringTypeBinaryLcase`.

I think that `isLowercaseCollation` is a very special edge case that shouldn't be baked in deeply in `Collation` object. `isBinary` is an actual property of collation and it makes sense to be here, but `isLowercaseCollation` is not.